SFR1.py

```diff
@@ -0,0 +1,228 @@
+        length_bounds (Union[tuple, int]): Sequence length interval to filter, default is (0, 2**32).
+        quality_threshold (int): Threshold value of average read quality (phred33) to filter, default is 0.
+
+    Return:
+        str: Path for filtered file.
+    """
+    records = list(SeqIO.parse(input_path, "fastq"))  # 1 элемент списка = 1 последовательноть формата fastq
+
+    filtered_records = []
+
+    # Обрабатываем границы ГЦ-состава
+    if isinstance(gc_bounds, (float, int)):
+        gc_min = 0
+        gc_max = gc_bounds
+    else:
+        gc_min = gc_bounds[0]
+        gc_max = gc_bounds[1]
+
+    # Обрабатываем границы по длине
+    if isinstance(length_bounds, int):
+        length_min = 0
+        length_max = length_bounds
+    else:
+        length_min = length_bounds[0]
+        length_max = length_bounds[1]
+
+    for record in records:
+        gc_content = calculate_gc_content(record.seq)
+        length = len(record)
+        avg_quality = calculate_average_quality(record)
+
+        if (
+            gc_min <= gc_content <= gc_max
+            and length_min <= length <= length_max
+            and avg_quality >= quality_threshold
+        ):
+            filtered_records.append(record)
+
+    return save_records_to_fastq(filtered_records, output_filename, input_path)
+
+
+def save_records_to_fastq(filtered_records: List[SeqIO.SeqRecord], output_filename: str, input_path: str) -> str:
+    """
+    Save filtered records to a new FASTQ file.
+
+    Arguments:
+        filtered_records (List[SeqIO.SeqRecord]): List of filtered SeqRecords.
+        output_filename (str): Name of the file to save.
+        input_path (str): Path to the original FASTQ file.
+
+    Returns:
+        str: Path to the saved file.
+    """
+    if output_filename == '':  # если имя не передано, используем имя входного файла
+        output_filename = os.path.basename(input_path)
+    if not output_filename.endswith(".fastq"):
+        output_filename += ".fastq"
+    output_dir = 'fastq_filtrator_results'  # задаём имя папки для сохранения результата
+    os.makedirs(output_dir, exist_ok=True)  # создаем папку, если она не существует
+    output_path = os.path.join(output_dir, output_filename)  # задаем путь, по которому сохранится файл
+
+    SeqIO.write(filtered_records, output_path, "fastq")
+
+    return output_path
+
+
+class BiologicalSequence(ABC):
+
+    @abstractmethod
+    def __len__(self) -> int:
+        pass
+
+    @abstractmethod
+    def __getitem__(self, index) -> str:
+        pass
+
+    @abstractmethod
+    def __str__(self) -> str:
+        pass
+
+    @abstractmethod
+    def alphabet_is_valid(self) -> bool:
+        pass
+
+
+class NucleicAcidSequence(BiologicalSequence):
+    def __init__(self, sequence: str):
+        self.sequence = sequence
+
+    def __len__(self) -> int:
+        return len(self.sequence)
+
+    def __getitem__(self, index) -> str:
+        return self.sequence[index]
```

Comment:
Можно было чуть проще:
```suggestion
    def __getitem__(self, index) -> str:
        pass
```